sktime/classification/distance_based/_time_series_neighbors.py

```diff
@@ -114,46 +111,68 @@ class KNeighborsTimeSeriesClassifier(_KNeighborsClassifier, BaseClassifier):
 
     _tags = {
         "capability:multivariate": True,
+        "X_inner_mtype": ["pd-multiindex", "numpy3D"],
         "classifier_type": "distance",
     }
 
     def __init__(
         self,
         n_neighbors=1,
         weights="uniform",
+        algorithm="brute",
         distance="dtw",
         distance_params=None,
-        algorithm="brute",
+        distance_mtype=None,
         leaf_size=30,
         n_jobs=None,
     ):
         self.n_neighbors = n_neighbors
         self.weights = _check_weights(weights)
+        self.algorithm = algorithm
         self.distance = distance
         self.distance_params = distance_params
-        if isinstance(self.distance, str):
-            distance = distance_factory(metric=self.distance)
-
-        self.algorithm = algorithm
+        self.distance_mtype = distance_mtype
         self.leaf_size = leaf_size
         self.n_jobs = n_jobs
 
-        super(KNeighborsTimeSeriesClassifier, self).__init__(
+        # translate distance strings into distance callables
+        if isinstance(distance, str) and distance not in DISTANCES_SUPPORTED:
+            raise ValueError(
+                f"Unrecognised distance measure string: {distance}. "
+                f"Allowed values for string codes are: {DISTANCES_SUPPORTED}. "
+                "Alternatively, pass a callable distance measure into the constuctor."
+            )
+
+        self.knn_estimator_ = KNeighborsClassifier(
             n_neighbors=n_neighbors,
             algorithm=algorithm,
-            metric=distance,
-            metric_params=None,  # Extra distance params handled in _fit
+            metric="precomputed",
+            metric_params=distance_params,
             leaf_size=leaf_size,
             n_jobs=n_jobs,
+            weights=weights,
         )
-        BaseClassifier.__init__(self)
 
-        # We need to add is-fitted state when inheriting from scikit-learn
-        self._is_fitted = False
+        super(KNeighborsTimeSeriesClassifier, self).__init__()
 
-    def fit(self, X, y, **kwargs):
-        """Override fit is required to sort out the multiple inheritance."""
-        return BaseClassifier.fit(self, X, y, **kwargs)
+        # the distances in sktime.distances want numpy3D
+        #   otherwise all Panel formats are ok
+        if isinstance(self.distance, str):
+            self.set_tags(X_inner_mtype="numpy3D")
+        elif distance_mtype is not None:
+            self.set_tags(X_inner_mtype=distance_mtype)
```

Comment:
what about case where self.distance is a PairwiseTransformer? The docstring says "only set this [`distance_mtype`] if distance is not BasePairwiseTransformerPanel descendant"

Reply:
then it will still override `X_inner_mtype`, but the `sktime` transformer will convert it back. If the user insists on an unnecessary conversion, why not.

Reply:
(One reason I ask is because at the top of the class, the tag is: `"X_inner_mtype": ["pd-multiindex", "numpy3D"]`.  But it looks like that is always over-written by the loop above.)

Reply:
resolved-  happy with explanation.